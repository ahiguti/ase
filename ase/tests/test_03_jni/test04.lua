
dofile("javadynload.lua")

local loader = GetFileLoader("ASETestJava.jar")
local klass = loader.loadClass("ASETestJava.TestObj")
local cbklass = loader.loadClass("ASETestJava.CallbackInterface")
local obj = klass()
obj.SetInt(3)
local cb = {
  CallString = function(x, y, z)
    -- print(x)
    return z
  end
}
obj.SetCallback(cbklass(cb))
local r = obj.CallbackString(100000, "foo", "bar", "baz")
print(r)

