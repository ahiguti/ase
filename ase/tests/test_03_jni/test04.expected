CallbackString x=foo y=bar z=baz
CallbackString r=baz
baz
